### PR TITLE
Feature/add lombok and exception

### PR DESCRIPTION
**Pull request by gaydukov**

### CODE REVIEW

File: src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java [hunk 1]
```diff
@@ -103,8 +124,15 @@ void givenPatient_whenUpdate_thenStatus200() throws Exception {
     @DisplayName("delete patient returned status")
     void givenPatient_whenDeletePatient_thenStatus204() throws Exception {
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java:125]: camelCase only

File: src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java [hunk 2]
```diff
@@ -117,10 +145,24 @@ void givenPatient_whenDeletePatient_thenStatus204() throws Exception {
     void givenPatients_whenGetPatients_thenStatus200andPatientsReturned() throws Exception {
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java:145]: camelCase only

File: src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/DoctorServiceImplTest.java [hunk 3]
```diff
@@ -28,9 +28,24 @@
     void givenDoctors_whenGetAllDoctors_thenReturnAllDoctors() {
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/DoctorServiceImplTest.java:28]: camelCase only

File: src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/DoctorServiceImplTest.java [hunk 4]
```diff
@@ -102,17 +154,20 @@ void givenDoctor_whenUpdateDoctor_thenReturnDoctorAndVerifyDoctorUpdate() {
     @DisplayName("delete doctor")
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/DoctorServiceImplTest.java:154]: displayName for tests should display understandable test name for a person who did not see your code and functionality!

File: src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/PatientServiceImplTest.java [hunk 5]
```diff
@@ -28,9 +28,24 @@
     void givenPatients_whenGetAllPatients_thenReturnAllPatients() {
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/service/serviceImpl/PatientServiceImplTest.java:28]: camelCase

**Review by shelsoftensy — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -1,56 +1,56 @@
...
-			<artifactId>spring-boot-starter-test</artifactId>
-			<scope>test</scope>
-		</dependency>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.5.3</version>
+        <relativePath/> <!-- lookup parent from repository -->
```
[the remark this answers is not in the document]
Comment (gaydukov) [reply pom.xml:9]: indicates the path to the parent pom.xml In this case, this tag is optional

File: src/main/java/com/softensy/SoftensySpringBoot/handlerException/RestResponseEntityExceptionHandler.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler(value = {ResponseStatusException.class})
+    protected ResponseEntity<Object> handleConflict(ResponseStatusException ex, WebRequest request) {
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/handlerException/RestResponseEntityExceptionHandler.java:14]: variable names

File: src/main/java/com/softensy/SoftensySpringBoot/entity/Doctor.java [hunk 8]
```diff
@@ -1,108 +1,30 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import javax.persistence.*;
-import java.sql.Date;
+import java.time.LocalDate;
 
 @Entity
 @Table
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/entity/Doctor.java:13]: do we need this annotation?

File: src/main/java/com/softensy/SoftensySpringBoot/entity/Patient.java [hunk 9]
```diff
@@ -1,108 +1,29 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
 
 import javax.persistence.*;
-import java.sql.Date;
+import java.time.LocalDate;
 
 @Entity
 @Table
```
Comment (shelsoftensy) [inline src/main/java/com/softensy/SoftensySpringBoot/entity/Patient.java:13]: do we need it?

File: src/test/java/com/softensy/SoftensySpringBoot/controller/DoctorControllerTest.java [hunk 10]
```diff
@@ -115,14 +144,28 @@ void givenDoctor_whenDeleteDoctor_thenStatus204() throws Exception {
     }
 
     @Test
-    @DisplayName("get all doctors returned doctor end status")
-    void givenDoctors_whenGetDoctors_thenStatus200() throws Exception {
+    @DisplayName("get all doctors returned doctors end status")
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/DoctorControllerTest.java:147]: end ?

File: src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java [hunk 11]
```diff
@@ -35,10 +35,17 @@
 
     @Test
     @DisplayName("get patient by id returned doctor end status")
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java:37]: end?

File: src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java [hunk 12]
```diff
@@ -49,16 +56,23 @@ void givenId_whenGetPatient_thenStatus200andPatientReturned() throws Exception {
                 .andExpect(jsonPath("$.lastName").value("Ivanov"))
                 .andExpect(jsonPath("$.middleName").value("Ivanovich"))
                 .andExpect(jsonPath("$.doctorId").value("1"))
-                .andExpect(jsonPath("$.dateOfBirth").value("11-05-1987"))
+                .andExpect(jsonPath("$.dateOfBirth").value("12-05-1987"))
                 .andExpect(jsonPath("$.phoneNumber").value("12345"));
     }
 
     @Test
     @DisplayName("save patient returned doctor end status")
```
Comment (shelsoftensy) [inline src/test/java/com/softensy/SoftensySpringBoot/controller/PatientControllerTest.java:64]: end?

**Review by shelsoftensy — CHANGES_REQUESTED** [review]

**Review by shelsoftensy — APPROVED** [review]

**Pull request merged**